rest/endpoints/playback_info/playback_info.yml

```diff
@@ -26,12 +26,90 @@ paths:
   /user_profiles/{user_profile_id}/assets/{asset_type}/{asset_id}:
     put:
       description: |
-        * **Available since:** NOT YET AVAILABLE
-      summary: # TODO
+        * **Available since:** 2020-01 (BPLAT-12229)
+      summary: Create/update a playback info of a given asset for a given user profile
       tags:
         - Playback Info
       operationId: putPlaybackInfo
-      # TODO
+      parameters:
+        - in: path
+          name: user_profile_id
+          required: true
+          description: The identity of the user profile to create playback info.
+          schema:
+            type: string
+        - in: path
+          name: asset_type
+          required: true
+          description: The asset type of the playback info.
+          schema:
+            type: string
+            enum: ["recording", "catchup", "vod"]
+        - in: path
+          name: asset_id
+          required: true
+          description: The asset id of the playback info.
+          schema:
+            type: string
+
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "../../snippets/PlaybackInfo/PlaybackInfoEntity.yml#/PlaybackInfoEntity"
```

Comment:
This entity looks like a full `PlaybackInfoEntity`, as you would expect in a response. We don't want to require the clients to provide all this data, as much of it can and will be created or derived server-side. e.g. the request body doesn't need to contain user ID or profile ID, as we can figure those out from the path parameters, nor does it need to contain `last_watched_at` as that will always be the current timestamp. Let's require as little as possible from the client in the request as we can while still being JSON API compliant.

Reply:
updated